### DISCUSSION TITLE
Handling of Stationary Objects in Frigate 13 Beta 7

### DISCUSSION THREAD

**Discussion by twister36** [blakeblackshear/frigate#9133] [Ask A Question]
After upgrading to version 13 the way the beta is handling Stationary Objects is driving me nut, especially cars...

In my environment I have 2 cameras that monitor the driveway and street in front of my house. I have them set to monitor for cars, people, dogs and cats. Now with Frigate 12 I had zero issues with cars being parked on the street, as the system would eventually mark them as stationary and ignore them.

However with version 13, the system has two behaviors that I've noticed for recordings:
1. Will record non-stop until the object leaves. And, the recording is in the on-going event section and will give you a play back for the entire time the object was stationary.

OR,

2. Will Record until something blocks the object momentarily and then start a new recording.  For example a car parked across the street may get blocked temporarily by another car driving by. This will cause the recording to stop, but the system will see the stationary car again and start all over. 

For Snapshots, the system will give a snapshot every minute or so and fill up the events page with the same stationary object over and over again. 

Now, @NickM-27 informed me to use required_zones, which works to limit the recordings/snapshot to those zones.... however, if a car comes and parks in one of those required_zones or if you place a trash can in one of those required zones, the system starts to go nuts again.

Here is the config for one of the cameras, the "Driveway":
![image](https://github.com/blakeblackshear/frigate/assets/114322877/3aa67a04-07af-42d8-8335-13d6872b554d)

Here is a non-stop recording example when required_zones is not configured or someone parks in one of my required zones. The system will continue to record it non-stop. And, yes, the whole 34mins of the video is playable... if the car says there for 2 hours, I will have a 2 hour recording.
![image](https://github.com/blakeblackshear/frigate/assets/114322877/26296ed4-12b2-44f6-bb79-a58b707d7287)

Here is a snapshot example when required_zones is not configured or if a stationary item is placed in one of my required_zones. The system will over and over create snapshots for the same car(s):
![image](https://github.com/blakeblackshear/frigate/assets/114322877/f9336fd7-309f-4b9d-8ef7-ad22b37bb740)

Again, I know I can use required_zones to limit this issue... However, when the stationary item occurs in the required_zone, the problem of the version 13 handling stationary items seems to be an issue or I'm doing something terrible wrong :)

Looking for guidance here, or if this is a issue that need to be looking into.
Please let me know if you need more information.

**Comment by gadget-man** [comment]
I've noticed similar issues. I've also noticed that if a car is parked in one of the zones and a new car drives past, it will often detect the license plate on one of the parked cars, and the snapshot will be of one of the parked vehicles rather than the new vehicle that has entered the zone.  The 'event' for that new vehicle arriving covers the whole period since the (now parked) car originally arrived (though I'm not sure that's different from the previous version...?)

**Reply by twister36** [reply]
I heard that Frigate may be able to see license plates, but I didn't see anything in the docs about it. Are you telling the system to look for license plates?

**Reply by gadget-man** [reply]
I have Frigate+ with my first trained model. Once linked, you can add license_plate as an object type to track. Other than a few issues with Frigate selecting an image for snapshot a little too late, it’s excellent.

**Reply by twister36** [reply]
Ah, that's awesome.. I also paid for Frigate+... but I haven't had time to dive into it yet. Thanks for the tip, that will be next on my list.

**Comment by twister36** [comment]
Here is a view of the Debug for the "Driveway" camera with required_zones setup. In this view you can see each zone and you can also see that the system keeps highlighting the stationary cars across the street to evaluate them (green box):
![image](https://github.com/blakeblackshear/frigate/assets/114322877/9f694a10-4fcb-4e12-87b5-558ef05ead67)

Now, even with required_zones applied, if someone comes and parks where I marked the red X, because it's in my required_zone, the system will start recording/snapshotting the parked car over and over.

**Reply by twister36** [reply]
Here is an example of "on-going" recording for Stationary Objects.
The white Yukon car is the object tagged for this video and you can see by the timeline that it is a 1 hour 27 min recording of this car, fully playable, for an object that physically has not moved. 
![image](https://github.com/blakeblackshear/frigate/assets/114322877/3b63c373-5560-49b7-b914-54d55dbb482e)

However, the system keeps thinking that the stationary Yukon car has moved, noted by the timeline icons:
![image](https://github.com/blakeblackshear/frigate/assets/114322877/cb3ea015-b187-49da-a851-06f13a9c770c)

**Reply by NickM-27** [reply]
I think you should start by explaining how you want the system to work, it seems you have some misunderstandings on how Frigate is working here and the config is the cause for it not working the way you want it to

**Comment by NickM-27** [comment]
Is your intention truly to have events for every car that drives by on the street? If so then the recording should not be a problem.

An event will be ongoing as long as the object is visible and tracked by frigate and given that you have the default record config (from what you have shown) any recording segments that have an active object will be saved. If this is not what you want then you should not be making the street a required zone for example.

**Reply by twister36** [reply]
outside of how I want the system to work, I am questioning why the system continues to think a stationary item, not a small object (car) and in good lightning is being marked as moving. Which causes the general issue of endless recordings/snapshots.

Concerning my setup, I have 80% of my system working as I want it to... I'm trying to understand the stationary issue and if there are missing configuration/settings I need to try.

**Reply by NickM-27** [reply]
> I am questioning why the system continues to think a stationary item, not a small object (car) and in good lightning is being marked as moving. Which causes the general issue of endless recordings/snapshots.

That doesn't cause multiple snapshots, they are just overlays on the recording. 

I'm struggling to understand if you do or don't want recordings for cars that are in the neighbors driveway / street.

**Reply by twister36** [reply]
To answer, no I don't want recordings of the neighbors driveways, and I was able to stop that by using the required_zones.

The concern is about someone parking in my required_zones,. This will cause endless recording, as noted above.

For example, the red X is a required zone (yellow area - street zone) to record delivery trucks, etc. However if someone parks there for 2 hours, I will get a recording for 2 hours. If someone parks there overnight, I will get a recording for the entire night.... this is because the system doesn't see that the car is stationary, like it should. 
![image](https://github.com/blakeblackshear/frigate/assets/114322877/9e3921d0-54fd-4933-8f28-1726659387f9)

**Reply by NickM-27** [reply]
> this is because the system doesn't see that the car is stationary, like it should.

no, it is not because of that. Your own screenshot shows that it detects the object as stationary.

The recording config will keep any segment where any object is active, not just that particular object that entered the zone. So if that car is parked in the zone but a person is walking on the sidewalk that recording segment will be kept. Currently there is no way to specify only keeping recording segments that the object that created the event is active

**Reply by NickM-27** [reply]
if you only want events for when a car parks in front of the driveway then the zone should be considerably smaller, right now your config is saying that any object that enters that area is important

**Reply by twister36** [reply]
The GUI shows that it doesn't see it as stationary... and it recorded the White Yukon for over 1 hour 27 mins  in this clip.
You can see that it goes back and forth saying it's moving when it's not
![image](https://github.com/blakeblackshear/frigate/assets/114322877/df745d16-698f-46b2-bf74-8a570de51a3b)

**Reply by NickM-27** [reply]
> You can see that it goes back and forth saying it's moving when it's not

so it is detected as stationary, it can't go back and forth without being stationary.

This is something that we are looking to improve in 0.14, but my point is that

> this is because the system doesn't see that the car is stationary, like it should.

is an incorrect statement because it does detect that the object is stationary. 

and again this isn't going to directly affect what is recorded because regardless of that particular object, any object being active will cause the recording to be saved due to your config.

**Comment by blakeblackshear** [comment]
Just want to chime in here and point out that there were no changes to the way Frigate handles stationary objects in 0.13. If 0.12 was stopping detection for (aka losing track of) stationary objects, that was a failure in the intended functionality. The improved motion sensitivity is probably helping to ensure that these objects are no longer lost which was always the intended behavior.

**Reply by twister36** [reply]
@blakeblackshear - Thanks for the reply and insight. Are there any settings available to adjust the sensitivity of the motion detection?

**Reply by NickM-27** [reply]
yes https://deploy-preview-6262--frigate-docs.netlify.app/configuration/motion_detection

**Comment by gadget-man** [comment]
That all makes sense. Is there a way to “encourage” the licence plate to be the one that’s within the bounds of the moving vehicle for a longstanding event, rather than one of the stationary objects?

**Reply by NickM-27** [reply]
that is something that would need to be done in the backend, knowing which car the license plate belongs to when the plate is within multiple bounding boxes is a difficult challenge

**Reply by blakeblackshear** [reply]
Yea. I notice this with fedex/ups/amazon sub labels too. I have some ideas for ways to improve this in 0.14.
